src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -94,8 +118,49 @@ public VisualStudioProjectTracker(IServiceProvider serviceProvider)
             }
         }
 
+        private void WaitForScheduledForegroundAffinitizedActions()
+        {
+            AssertIsForeground();
+
+            lock (_gate)
+            {
+                _taskForForegroundAffinitizedActions.Wait();
```

Comment:
How does this work? If we're running this on the UI thread, and these tasks are scheduled to the UI thread, why won't this deadlock?